content/en/blog/_posts/2024-04-22-userns-beta/index.md

```diff
@@ -0,0 +1,156 @@
+kernel and able to access / modify files on the host or other containers. If we
+run our pods with user namespaces, the privileges the container has over the
+rest of the host are reduced and the files outside the container it can access
+are limited too.
+
+In Kubernetes v1.25 we introduced support for user namespaces for only stateless
+pods. Kubernetes 1.28 lifted that restriction and now, with Kubernetes 1.30, we
+are moving to beta!
+
+## What is a user namespace?
+
+Note: Linux user namespaces are a different concept from [kubernetes
+namespaces](/docs/concepts/overview/working-with-objects/namespaces/).
+The former is a Linux kernel feature, the latter is a kubernetes feature.
+
+User namespaces is a Linux feature that isolates the UIDs and GIDs of the
+containers from the ones on the host. The identifiers in the container can be
+mapped to identifiers on the host in a way where the host UID/GIDs used for
+different containers never overlap. Furthermore, the identifiers can be mapped
+to unprivileged non-overlapping UIDs and GIDs on the host. This brings two key
+benefits:
+
+ * _Prevention of lateral movement_: As the UIDs and GIDs for different
+containers are mapped to different UIDs and GIDs on the host, containers have a
+harder time attacking each other even if they escape the container boundaries.
+For example, if container A is running with different UIDs and GIDs on the host
+than container B, the operations it can do on container B's files and processes
+are limited: only read/write what a file allows to others, as it will never
+have permission owner or group permission (the UIDs/GIDs on the host are
+guaranteed to be different for different containers).
+
+ * _Increased host isolation_: As the UIDs and GIDs are mapped to unprivileged
+users on the host, if a container escapes the container boundaries, even if it
+is running as root inside the container, it has no privileges on the host. This
+greatly protects what host files it can read/write, which process it can send
+signals to, etc. Furthermore, capabilities granted are only valid inside the
+user namespace and not on the host, which also limits the impact a container
+escape can have.
+
+{{< figure src="/images/blog/2024-04-22-userns-beta/userns-ids.png" alt="Image showing IDs 0-65535 are reserved to the host, pods use higher IDs" title="User namespace IDs allocation" >}}
+
+
+Without using a user namespace, a container running as root in the case of a
+container breakout, has root privileges on the node. And if some capabilities
+were granted to the container, the capabilities are valid on the host too. None
+of this is true when using user namespaces (modulo bugs, of course 🙂).
+
+## Changes in 1.30
+
+In Kubernetes 1.30, besides moving user namespaces to beta, the contributors
+working on this feature:
+
+ * Introduced a way for the kubelet to use custom ranges for the UIDs/GIDs mapping 
+ * Enforce the runtime supports all the features needed for user namespaces,
```

Comment:
```suggestion
 * Enforced the runtime supports all the features needed for user namespaces,
```

Reply:
I don't think this is right. The previous item uses "introduced" (past) because adding that feature was done in the past. Here enforce refers to what Kubernetes *does*, in the present.

I rephrased it as:
```
 * Added a way for Kubernetes to enforce the runtime supports all the features
   needed for user namespaces. If they are not supported, Kubernetes will show a
   clear error when trying to create a pod with user namespaces. Before 1.30, if
   the container runtime doesn't support user namespaces, the pod could be created
   without a user namespace.
```

Let me know if you think this is better or if I'm missing something and indeed using "enforced" is correct and simpler.

Reply:
That seems clearer, thank you!